wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/service/TextDocumentService.java

```diff
@@ -306,6 +310,21 @@ public DocumentHighlight documentHighlight(TextDocumentPositionParamsDTO positio
     	}
     	return null;
     }
+    
+    @POST
+    @Path("codeAction")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+	public List<? extends Command> codeAction(CodeActionParamsDTO params)
+					throws LanguageServerException, InterruptedException, ExecutionException {
+    	params.getTextDocument().setUri(prefixURI(params.getTextDocument().getUri()));
```

Comment:
We're consequently calling five different methods in a one line here, is it intentional? I can assure such lines would make a lot of troubles when someone would try to read and understand them. I saw several places where source code is unreasonably complex, so I think that would be very nice of you if you could simplify it where it is appropriate and possible so other people could understand the code too.